ls1intum/Artemis
# `Iris`: Add enable/disable Iris when creating programming exercise

### Checklist
#### General


#### Client
#### Changes affecting Programming Exercises

### Motivation and Context


It is necessary to provide the enable/disable option of Iris when creating a programming exercise.
### Description

Add the 'Iris Chatbot' checkbox in the bottom of programming exercise creation page when Iris profile is active. Iris enabled/disabled information will also shown at programming exercise detail page. 
### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 programming exercises with Iris enabled and disabled

On test server 9 (iris test server):
1. Log in to Artemis
2. Navigate to Course Management
3. Create a programming exercise with Iris enabled and check the detail page if Iris enabled/disabled information is shown. Open the programming exercise and check if there is Iris chatbot widget at the bottom.
4. Create a programming exercise with Iris disabled and check.
On other test server, you won't see Iris related when creating programming exercise, but in exercise detail page, there will show 'Allow Iris Chatbot: false' anyway.

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage





### Screenshots


![image](https://github.com/ls1intum/Artemis/assets/60965100/a18d435a-08da-4bd3-84e2-91586bd15443)
<img width="240" alt="image" src="https://github.com/ls1intum/Artemis/assets/60965100/a05726c6-d13b-4774-ac66-8f025b014366">
<img width="939" alt="image" src="https://github.com/ls1intum/Artemis/assets/60965100/3bbb858e-2bc9-4ec8-bce4-7ef7b9ffe2ad">
